### ISSUE TITLE
Asynchronous verifyClient middleware causes client socket to hang up

### ISSUE THREAD

**Issue by rockmacaca** [uWebSockets/uWebSockets#338]
I spent a pretty decent amount of time tracing this one down until I realized uWebSockets was the issue. It may have something to do with the fact that it's behind a SocketCluster setup, but it seems unlikely.

* What operating system are you running?
  * Windows
* What platform/language are you using?
  * Node.js
* Are you using SSL, if so, how?
  * SSL is not part of the equation

To reproduce:
* Run `socketcluster create TestApp`
* Add a line in `worker.js` to add async handshake middleware, i.e.
```javascript
  scServer.addMiddleware(scServer.MIDDLEWARE_HANDSHAKE, (req, next) => setImmediate(next));
```
This should result it the errors below in Chrome:
![image](https://cloud.githubusercontent.com/assets/4993980/20473600/cf8e8694-af8f-11e6-9f41-36dd66d98d8e.png)

This issue is not present if `ws` is specified as the WebSocket backend for SocketCluster.

**Comment by alexhultman** [comment]
Thanks for reporting. This cannot be fixed currently, but depends on how nodejs camp decides to handle this. Probably it will never be fixed.

**Comment by rockmacaca** [comment]
Well, good to know it's only Windows-specific. We're going to be deploying on Linux in production, so at least I should be able to use uWS there where performance matters. Thanks so much for looking further into this.

P.S. Though the whole situation really sucks, that `splendid` tag gave me a pretty good laugh.

**Comment by alexhultman** [comment]
Hey thanks for sticking around, I first thought this was just a simple js bug that could be fixed by anyone but yeah its a pretty central issue on Windows.

**Comment by alexhultman** [comment]
Fixed in 24ad90c689fef123675f7a8cf9145a6e9f851371 should be in v0.13.0
